File: TcpSocketSystem_DotNet/src/UseCases/ProcesarEco/ProcesarEcoHandler.cs
using TcpSocketSystem.Core.Domain.Modelos;
using TcpSocketSystem.Core.Domain.Puertos;

namespace TcpSocketSystem.Core.UseCases.ProcesarEco;

/// <summary>
/// Implementación del Caso de Uso para procesar el eco de red.
/// </summary>
public sealed class ProcesarEcoHandler : IProcesarEcoUseCase
{
    private readonly IServicioLogging _logger;
    private readonly string _nombreServidor;

    public ProcesarEcoHandler(IServicioLogging logger, string nombreServidor = "Servidor-DotNet10")
    {
        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
        _nombreServidor = nombreServidor;
    }

    public Task<MensajeEco> EjecutarAsync(string contenidoRaw, string emisor, CancellationToken cancellationToken = default)
    {
        // Validar y sanitizar a través de la entidad de dominio
        var mensajeCliente = new MensajeEco(contenidoRaw, emisor);

        _logger.Info($"Procesando mensaje [{mensajeCliente.Id}] de {mensajeCliente.Emisor}: \"{mensajeCliente.Contenido}\"", nameof(ProcesarEcoHandler));

        // Regla de negocio del eco: Reenviar el contenido agregando marca de respuesta
        var respuestaEco = MensajeEco.CrearRespuestaEco(mensajeCliente, _nombreServidor);

        return Task.FromResult(respuestaEco);
    }
}

File: TcpSocketSystem_Java/src/com/tecnm/sockets/usecases/procesareco/ProcesarEcoHandler.java
package com.tecnm.sockets.usecases.procesareco;

import com.tecnm.sockets.domain.modelos.MensajeEco;
import com.tecnm.sockets.domain.puertos.IServicioLogging;

/**
 * Implementación del Caso de Uso para procesar mensajes de eco en Java.
 */
public final class ProcesarEcoHandler implements IProcesarEcoUseCase {

    private final IServicioLogging logger;
    private final String nombreServidor;

    public ProcesarEcoHandler(IServicioLogging logger, String nombreServidor) {
        if (logger == null) {
            throw new IllegalArgumentException("El logger no puede ser nulo.");
        }
        this.logger = logger;
        this.nombreServidor = (nombreServidor != null) ? nombreServidor : "Servidor-Java21";
    }

    @Override
    public MensajeEco ejecutar(String contenidoRaw, String emisor) {
        MensajeEco mensajeCliente = new MensajeEco(contenidoRaw, emisor);

        logger.info(String.format("Procesando mensaje [%s] de %s: \"%s\"",
                mensajeCliente.getId(),
                mensajeCliente.getEmisor(),
                mensajeCliente.getContenido()), "ProcesarEcoHandler");

        return MensajeEco.crearRespuestaEco(mensajeCliente, nombreServidor);
    }
}
